pilot/pkg/trustbundle/trustbundle.go

```diff
@@ -161,19 +177,103 @@ func (tb *TrustBundle) UpdateTrustAnchor(anchorConfig *TrustAnchorUpdate) error
 	return nil
 }
 
+func (tb *TrustBundle) updateRemoteEndpoint(spiffeEndpoints []string) {
+	tb.mutex.RLock()
+	remoteEndpoints := tb.endpoints
+	tb.mutex.RUnlock()
+
+	endpointChange := !isEqSpliceStr(spiffeEndpoints, remoteEndpoints)
+	if !endpointChange {
+		return
+	}
+	trustBundleLog.Infof("updated trustAnchor Remote Endpoint  :%v", spiffeEndpoints)
+
+	tb.mutex.Lock()
+	tb.endpoints = spiffeEndpoints
+	tb.endpointUpdateChan <- struct{}{}
+	tb.mutex.Unlock()
+}
+
 // AddMeshConfigUpdate : Update trustAnchor configurations from meshConfig
 func (tb *TrustBundle) AddMeshConfigUpdate(cfg *meshconfig.MeshConfig) {
+	var err error
 	if cfg != nil {
 		certs := []string{}
+		endpoints := []string{}
 		for _, pemCert := range cfg.GetCaCertificates() {
-			certs = append(certs, pemCert.GetPem())
+			cert := pemCert.GetPem()
+			if cert != "" {
+				certs = append(certs, cert)
+			} else if pemCert.GetSpiffeBundleUrl() != "" {
+				endpoints = append(endpoints, pemCert.GetSpiffeBundleUrl())
+			}
 		}
-		err := tb.UpdateTrustAnchor(&TrustAnchorUpdate{
+
+		err = tb.UpdateTrustAnchor(&TrustAnchorUpdate{
 			TrustAnchorConfig: TrustAnchorConfig{Certs: certs},
 			Source:            SourceMeshConfig,
 		})
 		if err != nil {
-			trustBundleLog.Errorf("failed to update meshConfig trustAnchors: %v", err)
+			trustBundleLog.Errorf("failed to update meshConfig PEM trustAnchors: %v", err)
+		}
+
+		tb.updateRemoteEndpoint(endpoints)
+	}
+}
+
+func (tb *TrustBundle) fetchRemoteTrustAnchors() {
+	var err error
+
+	tb.endpointMutex.RLock()
+	remoteEndpoints := tb.endpoints
+	tb.endpointMutex.RUnlock()
+	remoteCerts := []string{}
+
+	if tb.remoteCaCertPool == nil {
+		tb.remoteCaCertPool, err = x509.SystemCertPool()
```

Comment:
looks like this is always `x509.SystemCertPool`, so just create it in the constructor? If not, this needs a lock

Reply:
It is changed in the unit testing code. Didn't want to add a lock unnecessarily since unit tests modify this before starting the main goroutine that uses this

Reply:
You can still change it in the unit testing code if you set it in the constructor I think